### ISSUE TITLE
Chart Directive not found

### ISSUE THREAD

**Issue by parakhkunal** [valor-software/ng2-charts#30]
I'm getting an error when I try to load the ng2-charts directives
I've the charts.js package installed via npm.

Any help would be appreciated

**Comment by parakhkunal** [comment]
Do we have to inject any dependencies in the boot file?

**Comment by born2net** [comment]
same issue here... any resolution?

**Comment by valorkin** [comment]
https://github.com/valor-software/ng2-charts#quick-start
point number 2
if installed via npm, just change the route

**Comment by born2net** [comment]
sorry I dont understand, what do you mean change route?
tx for the reply,

Sean

**Comment by valorkin** [comment]
I mean in quickstart sample using bower
`<script src="bower_components/Chart.js/Chart.min.js"></script>`
if you will be using npm, it should be something line
`<script src="node_modules/chart.js/Chart.min.js"></script>`

**Comment by parakhkunal** [comment]
Hi valorkin,

I'm using node_modules and I have imported the right script. The error is not at the import statement. It only occurs when I provide CHART_DIRECTIVES as a part of the directives list.

**Comment by valorkin** [comment]
than provide more details about error
and maybe gist for your use case

**Comment by born2net** [comment]
tx again for the time, for me, I don't get any errors, and I do see the Series A and Series B but nothing is rendering.
You can see my webconfig:
https://github.com/born2net/studioEnterprise/blob/charts/webpack.config.js

and you can see my source:
https://github.com/born2net/studioEnterprise/blob/charts/src/comps/app1/business/BusinessView.ts

I am suspecting something is wrong with my webconfig as I am not loading the same way as you, but did include the CDN for charts.js

regards

**Comment by valorkin** [comment]
https://github.com/born2net/studioEnterprise/blob/charts/src/comps/ng2-charts/demo/index.html#L36

**Comment by valorkin** [comment]
I am afraid there could be a problem with charts.js itself
if you are (and you are) using `use strict`
chart.js wrapped in `(function(){}).call(this);`
with `use strict` this will be `undefined` and not windows
to make it work properly we need to update chart.js lib itself
to be compiled in UMD style at least
so you can require it

**Comment by born2net** [comment]
tx can you recommend snippet? I think if I can include it in http://ng2.javascriptninja.io/ it will add lots of followers to this project,,,

**Comment by born2net** [comment]
I had some progress but now stuck in Uncaught ReferenceError: webpackJsonp is not defined

**Comment by parakhkunal** [comment]
@valorkin ,
Here is the gist link [https://gist.github.com/parakhkunal/acabbc452ed9a8a7d89f]

I keep getting this error.
Uncaught SyntaxError: Unexpected token <
Uncaught SyntaxError: Unexpected token <
	Evaluating http://localhost:3002/ng2-charts/ng2-charts
	Error loading http://localhost:3002/app/boot.js

**Comment by cdaniel77** [comment]
@parakhkunal, I am getting that exact error now.  Have you resolved the issue?

**Comment by parakhkunal** [comment]
@cdaniel77 , I could not. Waiting for some help from @valorkin

**Comment by born2net** [comment]
most of the issues were with my webconfig, this works:
https://github.com/born2net/studioEnterprise/blob/charting/webpack.config.js

and here main dir:
https://github.com/born2net/studioEnterprise/tree/charting/src/comps

but while I was able to load everything without errors, I could not get it to render.
for now I am using ng2-highcharts until this project becomes more stable.

tx all for help

Sean

**Comment by cdaniel77** [comment]
@born2net , do you have an example I can follow?  I can't seem to find anything past the initial install for high-charts for angular 2 beta.  The GitHub for it has me very confused.

**Comment by born2net** [comment]
I have the links above, and a "smaller" project, but based on same git:
https://github.com/born2net/ng2-charts
but honestly I could not get it to render so not sure if worth your time...
for now I went with: https://github.com/Bigous/ng2-highcharts which works in 5min

I am sure this project will get better and more stable as all ng2 stuff in all beta...

**Comment by inoabrian** [comment]
I figured it out! =)

1)   It looks like they wanted to update to use CHART_DIRECTIVES but in the `node_modules/ng2-charts/components/charts/charts.ts` they are exporting this 
`export const charts:Array<any> = [Charts, BaseChart];`
so I changed it to this 
`export const CHART_DIRECTIVES:Array<any> = [Charts, BaseChart];`

That solved this problem in my component
`import {CHART_DIRECTIVES} from 'ng2-charts/ng2-charts';`

2)   I also added a file to node_modules/ng2-charts

```
ng2-charts.ts
with export * from '.components/charts/charts';
```

**Comment by cdaniel77** [comment]
@inoabrian , those changes are already in my ng-charts code.  My export has the exact same thing you have but it still throws the error that @parakhkunal has.

**Comment by cdaniel77** [comment]
(ng-charts:1) Uncaought SyntaxError: Unexpected token <
(angular-polyfills.js:138) Uncaught SyntaxError: Unexpected token <
Evaluating http://loclahost:3000/ng2-charts/ng2-charts
Error Loading http://localhost:3000/app/boot.js

**Comment by cdaniel77** [comment]
@inoabrian , did you add the map inside packages or outside?

**Comment by inoabrian** [comment]
@cdaniel77 sorry, it actually kept breaking in different areas when I got that piece to work... =/
And I tried ng2-highcharts but couldn't get that one to work either..

**Comment by cdaniel77** [comment]
the only thing I noticed is that when I add a new folder outside of the app folder, if I don't set the defaultExtensions in the index.html I get the exact same error.  Once I add the new folder to the System.config and set it's defaults then the error goes away.  I am wondering if the chart.js folder itself should be outside the node_modules and set in the System.config but leave the ng2-charts folder in the node_modules.  If that could be a possible fix.

**Comment by inoabrian** [comment]
I couldn't event get the demo working locally. Too many deprecated dependencies.

**Comment by parakhkunal** [comment]
@cdaniel77 According to me, it's best to keep all the node dependency modules inside the node_modules directory.  Were you able to see the charts after your fix?

**Comment by inoabrian** [comment]
Nope nothing.. Is yours up and running @parakhkunal

**Comment by parakhkunal** [comment]
@inoabrian No luck for me either

**Comment by inoabrian** [comment]
Actually I get the bar example up without errors.... But the chart does not render I just get two <li> tags 
. Series A
. Series B

I'm looking into it now

**Comment by inoabrian** [comment]
I got it to work...
Follow this guy's stack Overflow post.
http://stackoverflow.com/questions/34653045/third-party-js-in-angular2-typescript

**Comment by cdaniel77** [comment]
I don't follow where things go at all.  So do we add a chart.d.ts file in the chart.js folder?  Can you provide more information please

**Comment by inoabrian** [comment]
chart.d.ts
Can go anywhere you want the definition files to be..
I have mine my tsd directory.

/app
--/home
--/schedule
/tsd
--/chartjs
-----/chart.d.ts
--/jquery
-----/jquery.d.ts

and in your new directive like in the stack overflow answer you have the reference to it via 
`/// <reference path="../../../../tsd/chartjs/chart.d.ts" />`

The new directive could be a shared component so I placed it in my common/components directory.
```
/// <reference path="../../../../tsd/chartjs/chart.d.ts" />
import {Directive, ElementRef, Renderer, Input} from 'angular2/core';
@Directive({
    selector: '[chart]',
    properties:[
      'chartData'
    ]
})
export class LineChartDirective {
    public data:any = {};
    private ctx:any = {};
    private lineChart:any = {};
    constructor(el: ElementRef, renderer: Renderer) {
        this.ctx = el.nativeElement.getContext("2d");
        this.lineChart = new Chart(this.ctx);
    }

    @Input() set chartData(data:any){
      if(!!data){
        this.data = data;
        this.lineChart.Line(this.data,{
          responsive: true
        });
      }
    }
}
```

And in my safety component where I want to render my chart
I import the LineChartDirective 

```
import {LineChartDirective} from '../common/Component/Charts/LineChart/linechart.directive';
@Component({
  selector: 'safety',
  template: `
    <div class="col-lg-12 col-md-12 col-sm-12 col-xs-12">
      <input [(ngModel)]="id" (keydown.enter)="getSingleElement()"class="form-control" type="text" required>
      <div class="col-lg-12 col-md-12 col-sm-12 col-xs-12">
        <canvas id="myChart" chart [chartData]="data" width="400" height="400"></canvas>
      </div>
    </div>
  `,
  directives: [NgClass, CORE_DIRECTIVES, FORM_DIRECTIVES,LineChartDirective]
})
```

**Comment by valorkin** [comment]
should be fixed in #136 please check (release 1.0.2)
